completions/ssh

```diff
@@ -451,20 +469,25 @@ _comp_xfunc_ssh_scp_remote_files()
         path=$(ssh -o 'Batchmode yes' "$userhost" pwd 2>/dev/null)
     fi
 
+    local escape_replacement='\\\\\\&'
+    if [[ ${less_escaping-} ]]; then
+        escape_replacement='\\&'
+    fi
+
     local files
-    if [[ ${1-} == -d ]]; then
+    if [[ ${dirs_only-} ]]; then
```

Comment:
Once it is initialized, `$dirs_only` is fine. But as far as we use the approach with `true`/`false`, one should just say `"$dirs_only"`.

```suggestion
    if "$dirs_only"; then
```

Similar for `less_escaping`.

Reply:
Do we need to escape `"$dirs_only"`, if we always initialize it? Seems like this would work as well:
```
if $dirs_only; then
```

Reply:
In case `IFS=abcdefghijklmnopqrstuvwxyz`

Reply:
Anyway, we are changing ourselves to quote similar cases in a context-independent manner.

Reply:
Makes sense. I'm not accustomed to thinking about all the crazy variations IFS that might happen :)

Reply:
Thanks!